audit/src/main/java/gov/cms/ab2d/audit/cleanup/FileDeletionServiceImpl.java

```diff
@@ -52,11 +59,25 @@ public void deleteFiles() {
         validateEfsMount();
 
         try (Stream<Path> filePaths = Files.walk(Paths.get(efsMount), FileVisitOption.FOLLOW_LINKS)) {
-            filePaths.filter(Files::isRegularFile).forEach(this::deleteFile);
+            List<Path> validFiles = filePaths.filter(Files::isRegularFile).collect(Collectors.toList());
+            // The list of jobs that were expired
+            Set<String> jobsDeleted = new HashSet<>();
+            for (Path file : validFiles) {
+                // Get the JobId from the name
+                var jobUuid = new File(file.toUri()).getParentFile().getName();
+                // See if the file should be deleted
+                boolean deleted = deleteFile(file);
+                if (deleted) {
+                    // If it was deleted
+                    jobsDeleted.add(jobUuid);
+                }
+            }
+            for (String job : jobsDeleted) {
```

Comment:
Not a huge deal but why do you need an extra for loop here instead of logging in the if statement above?

Reply:
I only adds jobs whose files are deleted because those are the only jobs that are done. If you call deleteFile it doesn't necessarily delete the file. It checks to see if it's old enough and if it isn't, it doesn't delete it. I didn't want to log jobs whose files were deleted in that process.

Reply:
Wouldn't it then make sense to log it in the if statement since you know at that point it was deleted? (and you'd avoid having to maintain the list).

Reply:
That would result in the multiple aggregate events being created for the same job since there could be multiple files deleted for the same job. jobsDeleted is a Set so it deduplicates.